src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -310,12 +353,31 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
         folder: vscode.WorkspaceFolder | undefined,
         config: AwsSamDebuggerConfiguration,
         token?: vscode.CancellationToken
-    ): Promise<undefined> {
-        const resolvedConfig = await this.makeConfig(folder, config, token)
-        if (!resolvedConfig) {
-            return undefined
+    ): Promise<void> {
+        try {
+            if (config.invokeTarget.target === 'api') {
+                await telemetry.apigateway_invokeLocal.run(async span => {
+                    const resolved = await this.makeConfig(folder, config, token)
+                    span.record({ httpMethod: resolved.api?.httpMethod })
+
+                    return this.invokeConfig(resolved)
+                })
+            } else {
+                await telemetry.lambda_invokeLocal.run(async () => {
+                    const resolved = await this.makeConfig(folder, config, token)
+
+                    return this.invokeConfig(resolved)
+                })
+            }
+        } catch (err) {
+            if (err instanceof SamLaunchRequestError) {
+                err.showNotification()
+            } else if (err instanceof ToolkitError) {
+                new SamLaunchRequestError(err.message, { ...err }).showNotification()
+            } else {
+                SamLaunchRequestError.chain(err, 'Failed to run launch configuration').showNotification()
```

Comment:
since the error no longer reaches vscode, do we need to call `vscode.debug.stopDebugging()` explicitly? https://github.com/microsoft/vscode/issues/102037

Reply:
We have to regardless as our whole integration is effectively a hack. We are launching debugger sessions during config resolution so we are responsible for terminating them appropriately: https://github.com/aws/aws-toolkit-vscode/pull/2865/files/5dca9e313f57ae439d7fc35c0e52e465174b6279#diff-b36cc56a594606340ec0320f08168751338535f63e948fdcaab177666baffb79R329-R332

Reply:
> while the debugger session is delegated to the appropriate extension per-language.

👍  So `vscode.debug.stopDebugging()` must be called by the debugger provider, not us.